src/exts/discord_integration.js

```diff
@@ -0,0 +1,94 @@
+const { bridge } = require('../spring_api');
+const { log } = require('../spring_log');
+const { config } = require('../launcher_config');
+const DiscordRPC = require("discord-rpc");
+
+if (config) {
+    if (config.discord_rich_presence.application_id == null) {
+        log.warn("config.discord_rich_presence.application_id not defined");
+        return
+    }
+}
+
+// Application ID from https://discord.com/developers/applications/<applicationId>/
+const applicationId = "1185990483143569448";
+const rpc = new DiscordRPC.Client({ transport: 'ipc' });
+
+DiscordRPC.register(applicationId);
+
+bridge.on('DiscordSetActivity', async command => {
+    // command
```

Comment:
maybe also add link to https://discord.com/developers/docs/rich-presence/how-to#updating-presence-update-presence-payload-fields there it's actually explained what those mean, and not just what type it is, I see you add comment to each assignment below, but you could just add them here instead and make it more concise.

Reply:
Will do